Kickstarter-iOS/Features/LoginTout/Controller/LoginToutViewController.swift

```diff
@@ -411,6 +411,26 @@ public final class LoginToutViewController: UIViewController, MFMailComposeViewC
     }
   }
 
+  fileprivate func createAuthorizationSession() -> ASWebAuthenticationSession? {
+    return OAuth.createAuthorizationSession { [weak self] result in
+      switch result {
+      case .loggedIn:
+        self?.viewModel.inputs.environmentLoggedIn()
```

Comment:
I went back-and-forth on where to put which pieces of logic. I settled on having the actual login call live in `OAuth.swift`, but the view controller is responsible for updating the UI when the login is complete.